platform/linux-generic/pktio/socket.c

```diff
@@ -696,19 +696,23 @@ static int sock_mmsg_recv(pktio_entry_t *pktio_entry, int index ODP_UNUSED,
 static uint32_t _tx_pkt_to_iovec(odp_packet_t pkt,
 				 struct iovec iovecs[MAX_SEGS])
 {
-	uint32_t pkt_len = odp_packet_len(pkt);
-	uint32_t offset = odp_packet_l2_offset(pkt);
```

Comment:
A simpler fix would seem to be to just initialize `offset` to 0 here, which is by definition the start of the packet regardless of its structure. It's also more efficient than the segment-based APIs, at least in linux-generic.

Reply:
Very true, thanks. Will send V2.